### ISSUE TITLE
Discussion: gracefully upgrading

### ISSUE THREAD

**Issue by ramontayag** [rails-api/active_model_serializers#1005]
Apologies if this doesn't belong here. I couldn't find a mailing list, and this does not seem to fit StackOverflow.

I'm working on a bunch of apps that use AMS `0.8.x` to `0.9.x`. Upgrading to `0.10.x` will break the API endpoints of my apps. How do you recommend we upgrade gracefully? An example: create a fork of AMS with a different namespace (e.g. `ActiveModel::SerializersOld`). The `v1` API endpoints of the apps will use this forked gem. Then, create `v2` endpoints that uses the new `ActiveModel::Serializers` `0.10.x` gem.

**Comment by bf4** [comment]
Maybe we could make adapters?  Some of this discussion will be addressed in the docs #994

**Comment by shicholas** [comment]
I think your idea of making serializers is great. Are you thinking something like a ``version_8`` adapter and a ``version_9`` one?

**Comment by shicholas** [comment]
@ramontayag I took your advice and created https://github.com/shicholas/active_model_serializers/tree/0-8-with-namespacing. This way we can require both ``0.10`` and ``0.8`` in the same project with the later being ``ActiveModel::SerializerVersion8``.

**Comment by bf4** [comment]
Watch out for any behavior it mixes in when required esp the controller

**Comment by shicholas** [comment]
thanks for the heads up.

**Comment by ramontayag** [comment]
Yes I took a quick stab at it and there are issues with the stuff AMS injects in the controller

**Comment by shicholas** [comment]
I've been noticing the same thing, so I opted to create a controller concern around the code in https://github.com/rails-api/active_model_serializers/issues/1002#issuecomment-121143172. @bf4, is anyone working on a version_8 adapter for version ``0.10``? I can take a stab at it

**Comment by joaomdmoura** [comment]
Hey you all! sorry for take so long to reply, I'm on a kind of business trip :smile:
So, first of all, enable people to upgrade from older version is definitely a priority for us! 
I totally agree that we should address it on the new `docs` structure. Probably a guide from ppl upgrading from 0.8 and another from 0.9. what they will need to change in order to make it happen.
 
I'm not sure about the adapter think yet, I'm still thinking about it. I didn't want to bring some of the old logic to this new version, but we can discuss that with other members as well later. 
Anyway, the adapter is a valid solution and if we had and adapter, even if it was outside the project, we would still distribute it trough the `docs` tutorial for sure!

@shicholas I'm almost sure there isn't anyone working on that so far. If you want to keep track of it make yourself comfortable to open a `WIP` PR.

**Comment by bf4** [comment]
Bump, if anyone wanna help with docs or code?

**Comment by bf4** [comment]
ref more code and docs in https://github.com/rails-api/active_model_serializers/pull/1290

**Comment by bf4** [comment]
ref custom adapters tutorial and discussion in https://github.com/rails-api/active_model_serializers/issues/1317

**Comment by bf4** [comment]
ref: mandatory issue https://github.com/rails-api/active_model_serializers/issues/1237

**Comment by bf4** [comment]
There's an issue to allow referring to serializers by string since it was allowed in a previous version https://github.com/rails-api/active_model_serializers/pull/1010

**Comment by remear** [comment]
ref https://github.com/rails-api/active_model_serializers/issues/600#issuecomment-53738222
